sympy/assumptions/cnf.py

```diff
@@ -402,6 +402,76 @@ def remove_literal(arg):
 class EncodedCNF:
     """
     Class for encoding the CNF expression.
+
+    Parameters
+    ==========
+    data : list of sets of integers
+        The CNF expression that has been encoded. Each set represents a clause
+        in the CNF expression, where the integers inside the set represent the
+        encoded symbols (variables) or their negations.
+
+    encoding : dict
+        A dictionary that maps symbols (variables) to their corresponding integer
+        encodings. The integers are used to represent the symbols in the 'data' list.
+
+    _symbols : list of symbols
+        A list containing the symbols (variables) that have been encoded. This list
+        is derived from the keys of the 'encoding' dictionary and may be useful for
+        other purposes or in future extensions of the class.
+
+        *Note that the order the symbols come in corresponds to their encoding.
+
+    Examples
+    ========
+
+    Create an instance of EncodedCNF and add a propositional expression.
+    >>> from sympy.assumptions.cnf import CNF, EncodedCNF
+    >>> from sympy.abc import a, b
+    >>> from sympy.logic.boolalg import Or, Not
+
+    >>> encoded_cnf = EncodedCNF()
+    >>> prop = Or(a, Not(b))
+    >>> encoded_cnf.add_prop(prop)
```

Comment:
I think it would be better if you made a CNF object and created the encoded CNF objected using that CNF object. In the existing code, whenever encoded cnf is used, this is how it's done.